xdp/src/packet.rs

```diff
@@ -0,0 +1,116 @@
+#![allow(clippy::arithmetic_side_effects)]
+
+use {libc::ETH_P_IP, std::net::Ipv4Addr};
+
+pub const ETH_HEADER_SIZE: usize = 14;
+pub const IP_HEADER_SIZE: usize = 20;
+pub const UDP_HEADER_SIZE: usize = 8;
+
+pub fn write_eth_header(packet: &mut [u8], src_mac: &[u8; 6], dst_mac: &[u8; 6]) {
+    packet[0..6].copy_from_slice(dst_mac);
+    packet[6..12].copy_from_slice(src_mac);
+    packet[12..14].copy_from_slice(&(ETH_P_IP as u16).to_be_bytes());
+}
+
+pub fn write_ip_header(packet: &mut [u8], src_ip: &Ipv4Addr, dst_ip: &Ipv4Addr, udp_len: u16) {
+    let offset = ETH_HEADER_SIZE;
```

Comment:
Can this consume a slice into which the header should be written? This would allow this to work with things like VLAN easier.